### ISSUE TITLE
Add integration tests for libraries

### ISSUE THREAD

**Issue by austin3dickey** [conbench/conbench#720]
This is a placeholder issue for what may be many issues in the future.

Because this is a monorepo and we build all our libraries during CI, we can, for example, test that `benchadapt` correctly publishes results to a running Conbench server during CI. This gives us better confidence than the unit tests in that library.

More to come.

**Comment by jgehrcke** [comment]
Yes! Most tests should use the client tooling we build for users. The Flask tests with the fake HTTP client use a testing approach that I have never liked. When I built API servers before I did even 'unit tests' with an _actual_ HTTP client and client tooling, ensuring conceptual isolation between _test runner_ and _tested system_. That isolation makes sure that tests focus on the right thing. We have many tests that use a combination of fake HTTP client and _direct_ database interaction -- hard to see through.

**Comment by austin3dickey** [comment]
As we discussed in recent PRs/issues, `benchalerts` has many unit tests that mock Conbench API responses, and "integration" (more like end-to-end) tests that hit live production servers. Both of these methods will have issues in CI if we change Conbench API responses. Implementing this issue correctly will be much safer and less brittle.

**Comment by austin3dickey** [comment]
This is becoming so important with rapid changes to the Conbench API endpoints that `benchalerts` uses, and also the changes to `benchalerts`'s usage itself. I need to set this up for `benchalerts` soon.
